lib/floe/workflow/intrinsic_function/transformer.rb

```diff
@@ -3,41 +3,189 @@
+          args = Transformer.process_args(args(), "States.ArrayContains", [Array, Object])
+          array, target = *args
+
+          array.include?(target)
+        end
+
+        rule(:states_array_range => {:args => subtree(:args)}) do
+          args = Transformer.process_args(args(), "States.ArrayRange", [Integer, Integer, Integer])
+          range_begin, range_end, increment = *args
+          raise ArgumentError, "invalid value for argument 3 to States.ArrayRange (given #{increment}, expected a non-zero Integer)" if increment.zero?
+
+          (range_begin..range_end).step(increment).to_a
+        end
+
+        rule(:states_array_get_item => {:args => subtree(:args)}) do
+          args = Transformer.process_args(args(), "States.ArrayGetItem", [Array, Integer])
+          array, index = *args
+          raise ArgumentError, "invalid value for argument 2 to States.ArrayGetItem (given #{index}, expected 0 or a positive Integer)" unless index >= 0
+
+          array[index]
+        end
+
+        rule(:states_array_length => {:args => subtree(:args)}) do
+          args = Transformer.process_args(args(), "States.ArrayLength", [Array])
+          array = args.first
+
+          array.size
+        end
+
+        rule(:states_array_unique => {:args => subtree(:args)}) do
+          args = Transformer.process_args(args(), "States.ArrayUnique", [Array])
+          array = args.first
+
+          array.uniq
+        end
+
+        rule(:states_base64_encode => {:args => subtree(:args)}) do
+          args = Transformer.process_args(args(), "States.Base64Encode", [String])
+          str = args.first
+
+          require "base64"
+          Base64.strict_encode64(str).force_encoding("UTF-8")
+        end
+
+        rule(:states_base64_decode => {:args => subtree(:args)}) do
+          args = Transformer.process_args(args(), "States.Base64Decode", [String])
+          str = args.first
+
+          require "base64"
+          begin
+            Base64.strict_decode64(str)
+          rescue ArgumentError => err
+            raise ArgumentError, "invalid value for argument 1 to States.Base64Decode (#{err})"
+          end
+        end
+
+        rule(:states_hash => {:args => subtree(:args)}) do
+          args = Transformer.process_args(args(), "States.Hash", [Object, String])
+          data, algorithm = *args
+          raise NotImplementedError if data.kind_of?(Hash)
+          if data.nil?
+            raise ArgumentError, "invalid value for argument 1 to States.Hash (given null, expected non-null)"
+          end
+
+          algorithms = %w[MD5 SHA-1 SHA-256 SHA-384 SHA-512]
+          unless algorithms.include?(algorithm)
+            raise ArgumentError, "invalid value for argument 2 to States.Hash (given #{algorithm.inspect}, expected one of #{algorithms.map(&:inspect).join(", ")})"
+          end
+
+          require "openssl"
+          algorithm = algorithm.sub("-", "")
```

Comment:
Very minor but we could have a static hash of their algo names mapped to `OpenSSL::Digest` algo names, could use .keys for the check above and avoid an array iteration here

Reply:
Yeah I can do that.  It turns out I only need this for Ruby 2.7 support - In Ruby 3.0 the full name just works.  Would you prefer the hash lookup, or maybe I can just gate this with a Ruby < 3.0 check?